scripts/verify-agent-memory.sh

```
@@ -73,6 +78,49 @@ for required_contract in "${required_agent_contracts[@]}"; do
 
+required_openrouter_contracts=(
+    '`OPENROUTER_API_KEY` is supplied by the environment.'
+    'if [[ -n "${OPENROUTER_API_KEY:-}" ]]; then'
+    'Run this check before asking Tim to create, paste, or reconfigure a key.'
+    'env -u OPENROUTER_API_KEY \'
+    '-u OPENAI_API_KEY \'
+    '-u OPENAI_BASE_URL \'
+    "uv run --with 'openai==3.1.0' --no-env-file --no-project python -c"
+    "'import openai; assert openai.__version__ == \"3.1.0\"'"
+    'Preparation is complete only when that import exits `0`.'
+    'openrouter_key=${OPENROUTER_API_KEY:?OPENROUTER_API_KEY is missing}'
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)